tests/ui/feature-gates/feature-gate-pin_ergonomics.rs

```diff
@@ -7,9 +7,13 @@ struct Foo;
 impl Foo {
     fn foo(self: Pin<&mut Self>) {
     }
+    fn foo_sugar(&pin mut self) {} //~ ERROR pinned reference syntax is experimental
+    fn foo_sugar_const(&pin const self) {} //~ ERROR pinned reference syntax is experimental
```

Comment:
@frank-king: As @compiler-errors had earlier mentioned, let's please add tests to confirm that this produces the desired error when the code is under `#[cfg(any())]`.